Lib/test/test_heapq.py

```diff
@@ -135,6 +153,13 @@ def test_heappushpop(self):
         x = self.module.heappushpop(h, 11)
         self.assertEqual((h, x), ([11], 10))
 
+    def test_heappop_max(self):
```

Comment:
What is the line being tested?  heappop_max() just calls heappop_internal() which is also used by the regular heappop().  I prefer testing the latter because it is part of the public API.  Likewise, the test should apply to both the C code and the pure python code (eventhough only the latter has a special case, the test should pass for both).   Alternatively, if you specifically want to exercise heappop_max(), it would be best to do it through the public function that uses it, merge().